corehq/middleware.py

```diff
@@ -113,11 +113,30 @@ def process_response(self, request, response):
 
 class TimeoutMiddleware(MiddlewareMixin):
 
+    @classmethod
+    def update_secure_session(cls, session, is_secure, domain=None):
```

Comment:
it looks like this method does two things, depending on `is_secure` - bumps the timeout, and toggles secure sessions.  I think it'd be clearer to separate the two.

Reply:
I tried this out but didn't especially like the result. I like that they're coupled because you shouldn't be calling them separately.